InsulinKit/PumpEvent.swift

```diff
@@ -119,16 +119,22 @@ extension PumpEvent {
 
             return DoseEntry(type: type, startDate: startDate, endDate: endDate, value: value, unit: unit, managedObjectID: objectID)
         }
-        set {
-            guard let entry = newValue else {
-                return
-            }
-
-            type = entry.type
-            startDate = entry.startDate as Date
-            endDate = entry.endDate as Date
-            value = entry.value
-            unit = entry.unit
+    }
+    
+    func setValuesWith(pumpEventType: PumpEventType?, dose: DoseEntry?) {
+        // This could also be done inline in DoseStore.addPumpEvents
+        // I'm OK with either place since it's only used once
```

Comment:
Fine with me. I kept this here because it was an adaption of the previous setter in the Event extension.